### ISSUE TITLE
Create Map directory in Nextcloud on Map creation

### ISSUE THREAD

**Issue by absurd-turtle** [ElektraInitiative/PermaplanT#572]
When a map is created by a user the backend should make a request to Nextcloud to create the directory hierarchy for the map.
The directory of the map should be named after the mapname in the backend.
nextcloud_root_dir
```
└── Maps
    ├── map_01
    │   ├── BaseLayer
    │   │   └── base_layer_image.png
    │   └── PhotoLayer
    │       ├── photo1.jpg
    │       └── photo2.jpg
    └── map_02
        ├── BaseLayer
        │   └── base_layer_image.png
        └── PhotoLayer
            ├── photo1.jpg
            └── photo2.jpg
```
Directories can be created by sending a MKCOL request to
remote.php/dav/files/user/path/to/new/folder

**Comment by markus2330** [comment]
@ThorbenStaufer can you tell us the place where this should be done ideally?

@absurd-turtle can you write the specific `curl` commands you would need to do for $USER? (Are all sub-dirs needed?)

Why not providing a [skeleton](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/default_files_configuration.html)? Then we could also add a few default images for the PhotoLayer as written in the use case? Or is the plan that a public share is used for thumbs up etc?

**Comment by ThorbenStaufer** [comment]
> can you tell us the place where this should be done ideally?

If we want to do it in the backend, the most logical place would be the map service.
Especially if we want to prevent any additions to the database, in case some error occurs in Nextcloud.

**Comment by absurd-turtle** [comment]
> @ThorbenStaufer can you tell us the place where this should be done ideally?
> 
> @absurd-turtle can you write the specific `curl` commands you would need to do for $USER? (Are all sub-dirs needed?)
> 
> Why not providing a [skeleton](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/default_files_configuration.html)? Then we could also add a few default images for the PhotoLayer as written in the use case? Or is the plan that a public share is used for thumbs up etc?

The problem is that the name of the map directories as well as the content there is individual for each user. I don't know if the skeleton can be of any help in this case.

**Comment by absurd-turtle** [comment]
> > can you tell us the place where this should be done ideally?
> 
> If we want to do it in the backend, the most logical place would be the map service. Especially if we want to prevent any additions to the database, in case some error occurs in Nextcloud.

I agree

**Comment by markus2330** [comment]
> The problem is that the name of the map directories as well as the content there is individual for each user. I don't know if the skeleton can be of any help in this case.

My idea was that we don't need to upload images but can simply copy them.

But you are right: if the common files are in public shared folders and only the user-specific files are in the above folders, the Skeleton cannot help. And probably that is the better way.

> `-u $USER:$PSW`

How would the auth in the backend work? Can we pass along some token?

Who creates `PermaplanT` and `PermaplanT/Maps`. Btw. we need to rename our `PermaplanT` folder maybe to `PermaplanTeam`?

**Comment by absurd-turtle** [comment]
> How would the auth in the backend work? Can we pass along some token?
Yes, the same token that is used for accessing data from the database can be used to request data from Nextcloud.

> Who creates `PermaplanT` and `PermaplanT/Maps`. Btw. we need to rename our `PermaplanT` folder maybe to `PermaplanTeam`?
I think this part could be done with the skeleton or default Nextcloud directories as those directories should be generated for each user. I don't know how to configure this in Nextcloud yet.

**Comment by markus2330** [comment]
related to collaboration
